app/client/ui/DocumentSettings.ts

```diff
@@ -328,12 +328,19 @@ export class DocSettingsPage extends Disposable {
         if (selected.get() === DocTypeOption.Regular) {
           docType = "";
         } else if (selected.get() === DocTypeOption.Template) {
-          docType = "Template";
+          docType = "template";
         } else {
-          docType = "Tutorial";
+          docType = "tutorial";
         }
         await persistType(docType, docId);
-        window.location.reload();
+        const currentUrl = window.location.href;
+        const urlId = currentUrl.split("/docs/")[1];
+        const cleanUrlId = urlId.match(/^doc\/.*/) ? urlId.split("/")[1] : urlId.split("/")[0];
+        const shortDocId = urlId?.split("~")[0];
+        const reloadUrl = urlId.match(/^doc\/.*/) ?
+          currentUrl.replace(`doc/${cleanUrlId}`, shortDocId) :
+          currentUrl.replace(cleanUrlId, shortDocId);
+        window.location.replace(reloadUrl ?? "");
```

Comment:
Not tested, but wouldn't this piece of code work? It should be enough and you would not be required to bring all the changes you made above:
```js
window.location.replace(urlState().makeUrl({
  docPage: undefined,
  fork: undefined, // will be automatically set once the page is reloaded
  doc: gristDocPageModel.currentDocId,
  // If that does not redirect to the page you want, you may play with Slug
  // See IGristUrlState: https://github.com/gristlabs/grist-core/blob/a5c9a494df93648e8110e19d351bd387cdf16be1/app/common/gristUrls.ts#L120
}));
```

Reply:
Thanks